### PR TITLE
SONARJAVA-4653 S6814 Optional REST parameters should have an object type

### PR DESCRIPTION
**Pull request by dorian-burihabwa-sonarsource**

### CODE REVIEW

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S6814.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "defaultSeverity": "Major",
+  "ruleSpecification": "RSPEC-6814",
+  "sqKey": "S6814",
+  "scope": "All",
+  "quickfix": "unknown",
+  "code": {
+    "impacts": {
+      "MAINTAINABILITY": "HIGH",
+      "RELIABILITY": "MEDIUM",
+      "SECURITY": "LOW"
```
Comment (johann-beleites-sonarsource) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S6814.json:19]: Do we expect that this has any impact at all on e.g. security? If yes, ignore this comment, if now, should we remove the security key?

File: java-checks/src/main/java/org/sonar/java/checks/spring/OptionalRestParametersShouldBeObjectsCheck.java
Comment (johann-beleites-sonarsource) [inline java-checks/src/main/java/org/sonar/java/checks/spring/OptionalRestParametersShouldBeObjectsCheck.java:1]: I might not understand some of the Spring specificities well enough here, but doesn't this also apply to [request parameters](https://www.baeldung.com/spring-request-param)?

**Review by johann-beleites-sonarsource — COMMENTED** [review]
The implementation looks good! I only have a couple of comments on the RSPEC (see the RSPEC PR) and a question about the scope of the rule.

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S6814.json (hunk 1 above)
Comment (dorian-burihabwa-sonarsource) [reply java-checks/src/main/resources/org/sonar/l10n/java/rules/java/S6814.json:19]: Indeed, I will remove that

File: java-checks/src/main/java/org/sonar/java/checks/spring/OptionalRestParametersShouldBeObjectsCheck.java
Comment (dorian-burihabwa-sonarsource) [reply java-checks/src/main/java/org/sonar/java/checks/spring/OptionalRestParametersShouldBeObjectsCheck.java:1]: Thanks, the annotation behavior seems similar enough that I should be able to add support pretty easily

**Review by johann-beleites-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**